dashboard/src/libs/useAllUsers.tsx

```diff
@@ -1,21 +1,20 @@
-import { createResource } from 'solid-js'
+import { createMemo, createResource, createRoot } from 'solid-js'
+import { User } from '/@/api/neoshowcase/protobuf/gateway_pb'
 import { client } from '/@/libs/api'
 
-/**
- * Create a all users resource:
- * ```typescript
- * const [users, { mutate, refetch }] = createAllUsersResource()
- * ```
- *
- * @returns ```typescript
- * [Resource<User[]>, { mutate: Setter<User[]>, refetch: () => void }]
- * ```
- */
-const useAllUsers = () => {
-  return createResource(async () => {
-    const allUsersRes = await client.getUsers({})
-    return allUsersRes.users
-  })
-}
+const [allUsersResource, { mutate: mutateAllUsers, refetch: refetchAllUsers }] = createResource(async () => {
+  const allUsersRes = await client.getUsers({})
+  return allUsersRes.users
+})
 
-export default useAllUsers
+export { allUsersResource, mutateAllUsers, refetchAllUsers }
+
+// keyにID, valueにユーザー情報を持つMap
+const allUsersMap = createRoot(() =>
```

Comment:
[ask] 単純な疑問ですが、ここでのcreateRootって必要なんですか？
色々調べたけどよくわからなかった

ここでも使ってるけど無くても動く
https://www.solidjs.com/tutorial/stores_nocontext

Reply:
うーんこういうことかな

> Creates a new non-tracked owner scope that doesn't auto-dispose. This is useful for nested reactive scopes that you do not wish to release when the parent re-evaluates.
https://www.solidjs.com/docs/latest#createroot

→ だからdispose functionを受け取る それを呼び出してdisposeする

> In essence this is a warning about a possibility of a memory leak due to a reactive computation being created without the proper context which would dispose of it when no longer needed.
A proper context is created a couple of different ways. Here are the ones I know about:
By using the render function.
By using the createRoot function. Under the hood render uses this.
By using the createContext function.
The first is by far the most common way, because each app has at least one render function call to get the whole show started.
https://stackoverflow.com/questions/70373659/solidjs-computations-created-outside-a-createroot-or-render-will-never-be

つまり今回のようにタブを閉じるまで必要なリソースの場合は特に必要ない...？
でもwarningは出ちゃうと

Reply:
無くても挙動は変わらない（そもそもdispose functionを受け取ってないし）けど、上のstackoverflowみたいな状況になって本当に対処すべきwarningが出たときにそれに気づけるように（狼少年にならないように）createRootでwarningを消しておく、というsolidjsのpracticeと捉えるのがいいかなー
それはコメントで書いておいてもよさそう

ちょっと気持ち悪いな

Reply:
ここコメントとかしてなくてごめんなさい
上のレビューにもある通り、`createRoot`つけてないと``computations created outside a `createRoot` or `render` will never be disposed``のwarningが出てたので`createRoot`つけてました

コメントで`createRoot`をツ買っている理由を書いておきます